### DISCUSSION TITLE
Administrative responsibilities

### DISCUSSION THREAD

**Discussion by derekcwhite** [stopthessotax/sso-wall-of-shame#10] [Announcements]
@daallgeier, @doransmestad, and I started this fork due to months-long lack of maintenance in the original repo, https://github.com/robchahin/sso-wall-of-shame/, despite several people making multiple attempts to contact the original creator to volunteer as maintainers.

I propose the following:

- No owner or member of this repo shall approve their own PR
- Should owners or members fail to approve, close, or respond to PRs in a reasonable timeframe, volunteers should be added as member
- A reasonable timeframe is a week
- To be added as a member, a volunteer must agree they will not approve their own PRs 

I hope there's not much else to add. The mission of sso.tax remains valid and in effect. Security should not be a luxury item.

**Comment by daallgeier** [comment]
Sounds reasonable to me, for sure.  Do we have pull requests from rob's repo?

**Comment by doransmestad** [comment]
Agreed.  Thank you for posting this!  I was thinking about this as well after I pushed direct to main yesterday.. :grimacing:  my apologies for that.  I'll use PRs moving forward!  :+1:

**Comment by derekcwhite** [comment]
@doransmestad I added branch protection rules to `main`, so it now requires an approval and can't be pushed to (though any of the three of us could allow pushes, push, and disallow pushes... but let's please not) :)

**Comment by mbainter** [comment]
Just as a note, I have multiple PRs sitting at a two-week mark with no review.

**Reply by doransmestad** [reply]
Apologies.  Thank you for the call out.  :+1:

**Comment by daallgeier** [comment]
On it.

**Reply by daallgeier** [reply]
Should we merge the changes to the original into ours too?

**Reply by doransmestad** [reply]
The original repo?  I'd like to, though I'm not going to be able to get to it this week  :(

**Comment by Snailson** [comment]
Is anyone still maintaining this project and resolving Issues?  Was going to submit one but it appears that they aren't getting closed.

**Reply by daallgeier** [reply]
Hi. I am still on here. One of my own pull requests haven't gone through due to me missing the actual pricing.

**Reply by daallgeier** [reply]
I was actually only looking out for Pulls Requests. That's my bad. Will check-in very soon.

**Reply by mbainter** [reply]
I am - but my workload right now hasn't permitted me to be very active. Should turn around in the near future.

**Comment by daallgeier** [comment]
Hi team, 
we need to find an easier way to get these things collaborated and improved upon.  

Do we have a template for the issues to enable us to create faster turn-around for issues to pull-requests?  

Should we have a bi-weekly check-in to collaborate on these issues and pull-requests as to be on top of them when they show up?  

Any other recommendations? 

I also just noticed that i had email notifications turned off and enabled them as to have faster turn around and feedback times too.  Will also use a chat bot too.

**Reply by mbainter** [reply]
There's [another PR](https://github.com/robchahin/sso-wall-of-shame/issues/266) suggesting adding more committers over on the upstream repo.  Might be worth inviting them to participate here, along with any other users that seem to be actively contributing there.

**Reply by jystadj** [reply]
A volunteer speaking up here. Fresh from another vendor ticking me off with their SSO tax.

I handle a lot of vendor selection and management in my day job, and I have always made it a point to call vendors out for this as often and thoroughly as possible.

Today I remembered sso.tax existed and realized I should contribute. I followed the fork over here when I realized Rob had gone quiet.

My workload is currently extra high, but I have a few vacation days coming up. I definitely have some more current information for the list, I'll try to get some PRs over in the next few days.

Happy to help out in other ways if you want.

**Comment by doransmestad** [comment]
Hey @mbainter @derekcwhite @daallgeier - hope you're well!  I'd like to suggest an update to the administrative policies here to help us keep moving forward.

At the start we aimed to avoid self-approval of pull requests, but I'd like to add a qualification to that.  I propose the following:   we will avoid self-approval of pull requests for a period of 3 days (72 hours) for simple changes (add/update/remove vendor), and a period of 7 days (168 hours) for more involved changes (refactoring, large PRs, etc).  If no other maintainer either approves, or indicates they wish to hold the PR for review further, during that time-frame, self-approvals are permitted.

Totally open to feedback/suggestion here - just want to give any of us the option to continue forward even if only one of us can actively participate at the time. 

Thoughts?

**Reply by doransmestad** [reply]
Quick update to this thread:  I'm going to proceed forward with self-approvals as described above, but if we wish to change that at any point, please let me know.  Thanks!

**Comment by derekcwhite** [comment]
Waiting 3 days before a self-approval seems like a reasonable approach imo!
